### ISSUE TITLE
mamba reshape/reduce

### ISSUE THREAD

**Issue by aguhaTT** [tenstorrent/tt-metal#7647]
problem statement:
```
C -> 1,1,32,32
hidden -> 1,1,32,163840

hidden @ C,
where, 
C permutes to 1,32,32,1
hidden reshapes to 1,32,5120,32 (this is very slow 34 ms, should be 0.01-0.1 ms)
```

two proposed solutions

**solution 1: bypass reshape**
replace matmul with mul and reduce
fast reduce exploration

**solution 2: fast reshape of hidden**
a modified form of the solution to #5428 

@kpaigwar @esmalTT @yan-zaretskiy

**Comment by aguhaTT** [comment]
![diagram for solution 2](https://github.com/tenstorrent/tt-metal/assets/159306286/07246d7f-555b-4fab-97ae-497321f43988)

**Comment by jliangTT** [comment]
My understanding is that the next step will be taken by mamba team, correct?

**Comment by apalaguha** [comment]
Yes, the next step will be taken by mamba team.

**Comment by esmalTT** [comment]
Closed by #7804
